client/utils/api.js

```diff
@@ -0,0 +1,28 @@
+import axios from 'axios';
+
+/**
+ * ApiCall - The base for all API call
+ * @param {object} data
+ * @param {string} type
+ * @param {string} url
+ * @returns {object}
+ */
+export default function apiCall(url, type, data,) {
+  return new Promise((resolve, reject) => {
+    const apiData = {
+      method: type,
+      url: url,
+      data: data
+    }
+    if(apiCall.data === undefined) {
+      apiData.delete;
```

Comment:
good stuff! @chibujax But please do you mean `delete apiData` instead of `apiData.delete`.

Reply:
Also, why do we delete apiData when `apiCall.data` === `undefined`@chibujax

Reply:
One more thing.. Line 17 should probably be `apiData`.data and not `apiCall`.data. @chibujax.

Reply:
It was more like a typo @tosmak16 , fixed. Thank you for spotting them out